File: src/test/mocks/responses/posts.response.ts
export const mockPostsResponse = () => ({
  posts: [
    {
      id: 1,
      title: 'Test Post 1',
      body: 'Test body 1',
      userId: 1,
      tags: ['test'],
      reactions: 0,
    },
    {
      id: 2,
      title: 'Test Post 2',
      body: 'Test body 2',
      userId: 2,
      tags: ['test'],
      reactions: 0,
    },
  ],
  total: 2,
  skip: 0,
  limit: 10,
});

File: app/src/main/java/com/example/androidkotlincompose/modules/home/posts/data/response/PostsResponse.kt
package com.example.androidkotlincompose.modules.home.posts.data.response

import kotlinx.serialization.Serializable

@Serializable
data class PostsResponse(
	val total: Int? = null,
	val limit: Int? = null,
	val skip: Int? = null,
	val posts: List<PostsItem?>? = null
)

@Serializable
data class Reactions(
	val dislikes: Int? = null,
	val likes: Int? = null
)

@Serializable
data class PostsItem(
	val reactions: Reactions? = null,
	val id: Int? = null,
	val title: String? = null,
	val body: String? = null,
	val userId: Int? = null,
	val views: Int? = null,
	val tags: List<String?>? = null
)
